### ISSUE TITLE
Can not reliably bind EIP to a fresh spot instance

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#3041]
_This issue was originally opened by @RomanSaveljev as hashicorp/terraform#17118. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version

```
Terraform v0.11.2
+ provider.aws v1.7.0
```

### Terraform Configuration Files

```hcl
provider "aws" {
    region = "eu-west-1"
    max_retries = "10"
}

resource "aws_spot_instance_request" "instance" {
    ami = "ami-94b236ed"
    spot_price = "0.8"
    wait_for_fulfillment = true
    spot_type = "one-time"
    instance_type = "m4.xlarge"
}

resource "aws_eip_association" "raas_windows_builder" {
    instance_id = "${aws_spot_instance_request.instance.spot_instance_id}"
    allocation_id = "eipalloc-27fc291a"
}
```

### Debug Output

https://gist.github.com/RomanSaveljev/7c2fff68cab944399859435f68ef7afc

### Expected Behavior

Terraform should have waited until spot instance is reported ready for binding elastic IP.

### Actual Behavior

Terraform attempted to bind EIP, but AWS has refused it.

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

Attached is a minimal testcase that reproduces the issue. EIP allocation needs to be your own.

**Comment by bflad** [comment]
I was able to reproduce this in an acceptance test and will submit a PR to fix it shortly.

**Comment by bflad** [comment]
This is merged into master and will be released with v1.8.0 of the provider.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.8.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#180-january-29-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by RomanSaveljev** [comment]
Thanks a lot. I have picked up new plugin version. Seems to work very well now
